pkg/kubelet/pod_workers.go

```diff
@@ -99,13 +101,25 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 		workQueue:                 workQueue,
 		resyncInterval:            resyncInterval,
 		backOffPeriod:             backOffPeriod,
+		podCache:                  podCache,
 	}
 }
 
 func (p *podWorkers) managePodLoop(podUpdates <-chan workUpdate) {
 	var minRuntimeCacheTime time.Time
+
 	for newWork := range podUpdates {
 		err := func() (err error) {
+			podID := newWork.pod.UID
+			if p.podCache != nil {
```

Comment:
I only used podCache if the container runtime is docker. This way, we were not blocked on  changes in rkt (which may no longer be needed). It also provides a simple knob to turn off podCache while testing.